Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -98,6 +98,15 @@ bool DecodeBool(pb_istream_t* stream) {
   }
 }
 
+void EncodeInteger(pb_ostream_t* stream, int64_t integer_value) {
+  return EncodeVarint(stream, google_firestore_v1beta1_Value_integer_value_tag,
```

Comment:
I didn't catch this on the first pass and we should defer it within the context of this PR this but there's an asymmetry in EncodeVarint and DecodeVarint. EncodeVarint also encodes the tag but Decode assumes the tag has already been decoded.

This is also present in EncodeBool/DecodeBool and while it's less obvious there I think that same lack of symmetry is potentially surprising.

Reply:
Huh; Good catch. I'll see if I can do something about that in a follow-up.